pkg/cmd/roachtest/test_runner.go

```diff
@@ -931,16 +931,9 @@ func (r *testRunner) runWorker(
 				}
 
 				// 50% chance of enabling the rangefeed buffered sender. Disabled by
-				// default. This is the reason we check for run time assertions rather
-				// than using rand:
-				// 1. The cluster setting RangefeedUseBufferedSender validation will fail
-				// if buildutil.CrdbTestBuild tag is not set.
-				// 2. There is a 50% chance that the cockroach binary used in roachtests
-				// includes the buildutil.CrdbTestBuild tag. And runtime assertion is
-				// enabled if and only if cockroach binaries used in roachtests include
-				// the buildutil.CrdbTestBuild tag.
-				useBufferedSender := roachtestutil.UsingRuntimeAssertions(t)
```

Comment:
`UsingRuntimeAssertions` was previously gating benchmark tests from enabling buffered sender. We should probably still disable buffered senders on benchmark tests? Unless you expect there to be no performance impact.

Reply:
We don’t anticipate any negative performance impact, and it would be good to verify if there’s any regression. If we feel strongly about this, the simplest approach might be to opt into this cluster setting when `registry.MetamorphicLeases` is enabled as that registry.TestSpec tries to achieve the same thing 
https://github.com/cockroachdb/cockroach/commit/fc68b0fe8d967ab48d96a659fd5a067787e5e734.

Reply:
My concern is less about negative regressions and more about increased variance in benchmark results. i.e. someone working on improving sysbench perf would need to both know about this metamorphic setting and make sure to conditionally enable/disable it or their results will fluctuate more than expected.

That being said I'm not familiar with this work at all so I'll defer to your judgement if you think the perf impact is minimal.

Reply:
I see, that's a good point. I would need to look closer at what the sysbench perf team is analyzing to provide a better answer. I would expect the perf impact to be minimal if the tests they are running don't enable `kv.rangefeed.enabled` but not sure otherwise.